tests/uhttp_ut/uhttp_ut.c

```diff
@@ -626,50 +615,50 @@ static void setup_uhttp_client_execute_request_with_content_mocks()
 
 static void setup_uhttp_client_dowork_no_msg_mocks()
 {
-    EXPECTED_CALL(xio_dowork(IGNORED_PTR_ARG));
-    STRICT_EXPECTED_CALL(singlylinkedlist_get_head_item(IGNORED_PTR_ARG));
-    STRICT_EXPECTED_CALL(singlylinkedlist_item_get_value(IGNORED_PTR_ARG));
-    STRICT_EXPECTED_CALL(BUFFER_length(IGNORED_PTR_ARG));
-    STRICT_EXPECTED_CALL(STRING_c_str(IGNORED_PTR_ARG));
+    STRICT_EXPECTED_CALL(xio_dowork(IGNORED_PTR_ARG));
+    STRICT_EXPECTED_CALL(singlylinkedlist_get_head_item(IGNORED_PTR_ARG)).CallCannotFail();
```

Comment:
>.CallCannotFail(); [](start = 73, length = 18)

Is this hiding something? It seems that if we do not fail this we would miss some paths? I am only flagging it to make sure that we do not miss something in the code (like a comparison with 0 instead of the a real enum value.

Reply:
The rest of the CallCannot Fail usages seem to be OK.

---
In reply to: [321915242](https://github.com/Azure/azure-uhttp-c/pull/40#discussion_r321915242) [](ancestors = 321915242)

Reply:
This has to not fail due to the test setup.  I need the dowork to always go through the list of items so they can get destroyed.  Normally this gets cleaned up in the destroy, but I can't call destroy in this test till the end.

---
In reply to: [321915370](https://github.com/Azure/azure-uhttp-c/pull/40#discussion_r321915370) [](ancestors = 321915370,321915242)